File: jimmer-ddd-java-spring-boot-starter/src/main/java/org/morecup/jimmerddd/java/spring/aggregateproxy/IdGenerator.java
package org.morecup.jimmerddd.java.spring.aggregateproxy;

import org.babyfish.jimmer.spring.cfg.support.SpringConnectionManager;
import org.babyfish.jimmer.sql.JSqlClient;
import org.babyfish.jimmer.sql.kt.KSqlClient;
import org.babyfish.jimmer.sql.runtime.JSqlClientImplementor;
import org.morecup.jimmerddd.core.aggregateproxy.BaseAssociatedFixedKt;
import org.springframework.jdbc.datasource.DataSourceUtils;

import java.sql.Connection;

public class IdGenerator {
    public static Object generatorId(JSqlClientImplementor jSqlClient, Class<?> entityType) {
        return generatorId(jSqlClient, entityType, null);
    }

    public static Object generatorId(JSqlClientImplementor jSqlClient, Class<?> entityType, Connection con) {
        if (con == null) {
            con = DataSourceUtils.getConnection(((SpringConnectionManager) jSqlClient.getConnectionManager()).getDataSource());
        }
        return BaseAssociatedFixedKt.commonGeneratorId(jSqlClient, entityType, con);
    }
}

File: Jimmer-ddd-kotlin-spring-boot-starter/src/main/kotlin/org/morecup/jimmerddd/kotlin/spring/aggregateproxy/IdGenerator.kt
package org.morecup.jimmerddd.kotlin.spring.aggregateproxy

import org.babyfish.jimmer.spring.cfg.support.SpringConnectionManager
import org.babyfish.jimmer.sql.kt.KSqlClient
import org.morecup.jimmerddd.core.aggregateproxy.commonGeneratorId
import org.springframework.jdbc.datasource.DataSourceUtils
import java.sql.Connection
import kotlin.reflect.KClass

fun generatorId(kSqlClient: KSqlClient,entityType: Class<*>,con: Connection? = null): Any? {
    val connection: Connection = con?:DataSourceUtils.getConnection((kSqlClient.javaClient.connectionManager as SpringConnectionManager).dataSource)
    return commonGeneratorId(kSqlClient.javaClient, entityType, connection)
}

fun generatorId(kSqlClient: KSqlClient,entityType: KClass<*>,con: Connection? = null): Any? {
    return generatorId(kSqlClient, entityType.java, con)
}